public/app/features/alerting/unified/enterprise-components/AI/AIGenAlertRuleButton/addAIAlertRuleButton.ts

```diff
@@ -0,0 +1,17 @@
+import { ComponentType, createElement } from 'react';
+
+export interface GenAIAlertRuleButtonProps {}
+
+// Internal variable to store the actual component
+let InternalAIAlertRuleButtonComponent: ComponentType<GenAIAlertRuleButtonProps> | null = null;
+
+export const AIAlertRuleButtonComponent: ComponentType<GenAIAlertRuleButtonProps> = (props) => {
+  if (!InternalAIAlertRuleButtonComponent) {
+    return null;
+  }
+  return createElement(InternalAIAlertRuleButtonComponent, props);
```

Comment:
I think we should also wrap all AI buttons into `ErrorBoundary`. This way we could ensure that the basic functionality keeps working if something goes wrong with the AI extensions.
If I'm not mistaken, currently, any issues in AI buttons would lead to full page crash